sui_programmability/examples/nft_standard/README.md

```diff
@@ -0,0 +1,114 @@
+    - We should have safe, convenient libraries for common royalty policies (e.g., no royalty, royalty split among set of addresses), commissions, and auctions
+- **Price discovery aka “NFT orderbook”**. The standards for listing NFT’s should provide (via event queries or other mechanisms) a common framework for discovering the price for a given NFT, the floor price of a collection, etc.
+
+### Implementation thoughts
+
+One approach to satisfying a number of these requirements is to define a shared `NFTSafe` that is associated with a specific address (similar to the `Safe` for coins proposed here: ‣).
+
+- The owner of the `NFTSafe` has an `OwnerCap` that allows them to mint `TransferCap`'s, which give the holder permission to transfer the NFT between two different `NFTSafe`'s.
+    - When an owner lists an NFT on a marketplace, they would actually list the `TransferCap`--the NFT continues to sit in the safe.
+    - A `TransferCap` doesn’t give the holder unconditional permission to transfer the NFT—they must also satisfy the royalty and commission policies. This is how royalty enforcement happens.
+    - If you have an `OwnerCap`, you can take an NFT out of the safe (i.e., make it a single-owner object), or transfer it to a different safe without paying royalties or commissions. This satisfies the “let a user transfer NFT between addresses” requirements
+- Borrowing an NFT can be implemented through a `BorrowCap` that is similar to `TransferCap`, but only allows the holder to temporarily take the NFT out of a safe during a transaction—they must return it before the transaction ends
+    - The “hot potato” design pattern is the natural implementation approach for this
+- An `OwnerCap` also allows the owner to
+    - transfer all of their NFT’s to a different user (by transferring the `OwnerCap`),
+    - gate access to NFT’s via a multisig (by putting the `OwnerCap` in a multisig smart contract wallet)
+    - Use [KELP](https://eprint.iacr.org/2021/289.pdf)-like social recovery schemes to protect the user against key loss (by putting the `OwnerCap` in a social recovery smart contract wallet)
+- NFT minting can either transfer the freshly created NFT directly to a user address, or put it into the user’s `NFTSafe`. Some users might want to allow arbitrary deposits of NFT’s into their safe, whereas others might only want to give themselves permission to pull an NFT into the safe (e.g., as a means of curation or spam prevention)
+- A single, heterogenous `NFTSafe` is probably the simplest/most convenient, but one could also imagine `NFTSafe<T>` that partitions by type. This draft PR goes for the latter.
+- A reasonable implementation of the `NFTSafe` idea will rely heavily on the forthcoming “dynamic child object access” feature: https://github.com/MystenLabs/sui/issues/4203.
+
+# Display
```

Comment:
For display standards, we discussed the idea of creating some trait-like behaviour. Relevant for the Sui NS project was the expiry of an NFT.

If an object has been recognized as some NFT, wallets can apply an `Expires` trait optionally. They do this by looking for `metadata.expiry` date (unix or ISO 8601?) or - if missing - then `metadata.is_expired` bool flag.
Any NFT can have expiry logic. By having this as a trait rather than on e.g. only domain name NFTs, we have a more composable standard.